src/priceHistory/priceHistoryChart.tsx

```diff
@@ -0,0 +1,357 @@
+            position={'top'}
+            chartWidth={chartWidth}
+          />
+        )
+        break
+
+      case lowestRateIdx:
+        result.push(
+          <ChartAwareSvgText
+            x={x}
+            y={y}
+            key={idx}
+            value={chartData[lowestRateIdx].displayValue}
+            position={'bottom'}
+            chartWidth={chartWidth}
+          />
+        )
+        break
+    }
+
+    switch (result.length) {
+      case 0:
+        return null
+      case 1:
+        return result[0]
+      default:
+        return <>{result}</>
+    }
+  }
+}
+
+interface PriceHistoryChartProps {
+  tokenId: string
+  containerStyle?: ViewStyle
+  testID?: string
+  chartPadding?: number
+  color?: colors
+}
+
+export default function PriceHistoryChart({
+  tokenId,
+  containerStyle,
+  testID,
+  chartPadding,
+  color = colors.black,
+}: PriceHistoryChartProps) {
+  const priceHistoryPrices = useSelector((state: RootState) =>
+    priceHistoryPricesSelector(state, tokenId)
+  )
+
+  const priceHistoryStatus = useSelector((state: RootState) =>
+    priceHistoryStatusSelector(state, tokenId)
+  )
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+  const localExchangeRate = useSelector(usdToLocalCurrencyRateSelector)
+  const dispatch = useDispatch()
+
+  const dollarsToLocal = useCallback(
+    (amount: BigNumber.Value | null) =>
+      convertDollarsToLocalAmount(amount, localCurrencyCode ? localExchangeRate : 1),
+    [localExchangeRate]
+  )
+  const displayLocalCurrency = useCallback(
+    (amount: BigNumber.Value) =>
+      getLocalCurrencyDisplayValue(amount, localCurrencyCode || LocalCurrencyCode.USD, true),
+    [localCurrencyCode]
+  )
+
+  useEffect(() => {
+    const latestTimestamp = priceHistoryPrices.at(-1)?.priceFetchedAt ?? 0
+    if (priceHistoryPrices.length > 0 && latestTimestamp > Date.now() - ONE_HOUR_IN_MILLIS) {
+      return
+    }
+
+    dispatch(
+      fetchPriceHistoryStart({
+        tokenId,
+        startTimestamp: Date.now() - ONE_DAY_IN_MILLIS * 30,
+        endTimestamp: Date.now(),
+      })
+    )
+  }, [])
+
+  if (priceHistoryStatus === 'loading' && priceHistoryPrices.length === 0) {
+    return <Loader />
+  } else if (priceHistoryPrices.length === 0) {
+    return null
+  }
+
+  // Create chart data from price history
+  const chartData = []
+  let lastTimestampAdded, highPrice, lowPrice
+  for (let i = 0; i < priceHistoryPrices.length; i++) {
+    // Check if price is the highest or lowest price and if so store it
+    if (!highPrice || priceHistoryPrices[i].priceUsd > highPrice.priceUsd) {
+      highPrice = priceHistoryPrices[i]
+    }
+    if (!lowPrice || priceHistoryPrices[i].priceUsd < lowPrice.priceUsd) {
+      lowPrice = priceHistoryPrices[i]
+    }
+
+    // Only grab one price per chart step & the most recent price
+    if (
+      lastTimestampAdded &&
+      priceHistoryPrices[i].priceFetchedAt - lastTimestampAdded <
```

Comment:
this assumes that the data is sorted by priceFetchedAt right?

Reply:
Yes which it should be from blockchain API. We could sort before to be sure, but I haven't tried to see how long this would take on the larger datasets.